File: frontend/src/services/escaneo.service.ts
import { api } from './api';
import { ApiResponse, Escaneo } from '../types';

export const listar = async (limit = 100): Promise<Escaneo[]> => {
  const { data } = await api.get<ApiResponse<Escaneo[]>>(`/escaneos?limit=${limit}`);
  return data.data!;
};

File: mobile/lib/services/escaneo_service.dart
import '../models/escaneo.dart';
import 'api_service.dart';

class EscaneoService {
  static Future<List<Escaneo>> listar({int limit = 100}) async {
    final data = await ApiService.get('/escaneos?limit=$limit') as List<dynamic>;
    return data.map((e) => Escaneo.fromJson(e as Map<String, dynamic>)).toList();
  }
}
